third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -737,6 +739,47 @@ LogicalResult PointerCanonicalizer::rewriteCondBranchOp(
   return success();
 }
 
+LogicalResult PointerCanonicalizer::rewriteSelectOp(arith::SelectOp selectOp,
+                                                    Location curLoc,
+                                                    OpOperand *curOperand,
+                                                    Value &nextPtr) {
+  const auto &operands = selectOp->getOpOperands();
+  bool otherIsFalse = &operands[1] == curOperand;
+  assert(otherIsFalse || &operands[2] == curOperand);
+
+  OpOperand *otherOperand = &operands[otherIsFalse ? 2 : 1];
+
+  // If we didn't traverse both operands, simply create the pointers
+  // for the operand we reached and save the fat pointer back. When (and if)
+  // the traversal meets the select again we can rewrite it properly
+  if (!pointers.contains(otherOperand->get())) {
+    FatPtr fatPtr = pointers[curOperand->get()];
+    Value tensorPtr = createTensorPointer(fatPtr, curLoc);
```

Comment:
The `createTensorPointer` is a bit confusing--I always read it as creating the full tensor pointer with offsets added. Maybe we should rename this as `splatBasePointer` to be exact?

Reply:
I read that as well. I now made it do exactly what was assumed by its name :)